mmdet/core/post_processing/bbox_nms.py

```diff
@@ -42,6 +42,12 @@ def multiclass_nms(multi_bboxes,
         tuple: (bboxes, labels, indices (optional)), tensors of shape (k, 5),
             (k), and (k). Labels are 0-based.
     """
+    if multi_bboxes.numel() == 0:
```

Comment:
@druzhkov-paul this error is not triggered in exportation but in evaluation (my model generates empty pred boxes as it couldn't learn anything). Ilya is looking into the issue I think you are referring to (https://jira.devtools.intel.com/browse/CVS-82705)

Reply:
> this error is not triggered in exportation

Are you sure? NMS is not called during the export of such a model.

> model generates empty pred boxes as it couldn't learn anything

What error is raised? Please provide some description of the failure.

Reply:
```
[ INFO ]  Epoch [1][15/15]	lr: 1.000e-02, eta: 0:03:07, time: 0.988, data_time: 0.279, memory: 7102, loss_rpn_cls: nan, loss_rpn_bbox: nan, loss_cls: nan, acc: 50.0000, loss_bbox: nan, loss_mask: nan, loss: nan
2022-03-30 11:37:55,910 - mmdet - INFO - Saving checkpoint at 1 epochs
[ INFO ]  Saving checkpoint at 1 epochs
Traceback (most recent call last):
  File "/opt/training_script/run.py", line 834, in <module>
    run()
  File "/opt/training_script/run.py", line 725, in run
    run_training(
  File "/usr/local/lib/python3.8/site-packages/base_logger/tools.py", line 180, in wrap_run_training
    result = _f(*args, **kwargs)
  File "/opt/training_script/run.py", line 198, in run_training
    training.train(dataset=dataset, output_model=output_model, train_parameters=train_parameters)
  File "/usr/local/lib/python3.8/site-packages/detection_tasks/apis/detection/train_task.py", line 125, in train
    train_detector(model=self._model, dataset=mm_train_dataset, cfg=training_config, validate=True)
  File "/usr/local/lib/python3.8/site-packages/mmdet/apis/train.py", line 253, in train_detector
    runner.run(data_loaders, cfg.workflow, compression_ctrl=compression_ctrl)
  File "/usr/local/lib/python3.8/site-packages/mmcv/runner/epoch_based_runner.py", line 127, in run
    epoch_runner(data_loaders[i], **kwargs)
  File "/usr/local/lib/python3.8/site-packages/detection_tasks/extension/utils/runner.py", line 64, in train
    self.call_hook('after_train_epoch')
  File "/usr/local/lib/python3.8/site-packages/mmcv/runner/base_runner.py", line 307, in call_hook
    getattr(hook, fn_name)(self)
  File "/usr/local/lib/python3.8/site-packages/mmdet/core/evaluation/eval_hooks.py", line 222, in after_train_epoch
    self._do_evaluate(runner)
  File "/usr/local/lib/python3.8/site-packages/mmdet/core/evaluation/eval_hooks.py", line 230, in _do_evaluate
    results = single_gpu_test(runner.model, self.dataloader, show=False)
  File "/usr/local/lib/python3.8/site-packages/mmdet/apis/test.py", line 33, in single_gpu_test
    result = model(return_loss=False, rescale=True, **data)
  File "/usr/local/lib/python3.8/site-packages/nncf/torch/dynamic_graph/wrappers.py", line 156, in wrapped
    return module_call(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/mmcv/parallel/data_parallel.py", line 42, in forward
    return super().forward(*inputs, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/torch/nn/parallel/data_parallel.py", line 165, in forward
    return self.module(*inputs[0], **kwargs[0])
  File "/usr/local/lib/python3.8/site-packages/nncf/torch/dynamic_graph/wrappers.py", line 156, in wrapped
    return module_call(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/mmcv/runner/fp16_utils.py", line 98, in new_func
    return old_func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/mmdet/models/detectors/base.py", line 219, in forward
    return self.forward_test(img, img_metas, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/mmdet/models/detectors/base.py", line 186, in forward_test
    return self.simple_test(imgs[0], img_metas[0], **kwargs)
  File "/usr/local/lib/python3.8/site-packages/mmdet/models/detectors/two_stage.py", line 207, in simple_test
    return self.roi_head.simple_test(
  File "/usr/local/lib/python3.8/site-packages/mmdet/models/roi_heads/standard_roi_head.py", line 258, in simple_test
    det_bboxes, det_labels = self.simple_test_bboxes(
  File "/usr/local/lib/python3.8/site-packages/mmdet/models/roi_heads/test_mixins.py", line 107, in simple_test_bboxes
    det_bbox, det_label = self.bbox_head.get_bboxes(
  File "/usr/local/lib/python3.8/site-packages/mmcv/runner/fp16_utils.py", line 186, in new_func
    return old_func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/mmdet/models/roi_heads/bbox_heads/bbox_head.py", line 315, in get_bboxes
    det_bboxes, det_labels = multiclass_nms(bboxes, scores,
  File "/usr/local/lib/python3.8/site-packages/mmdet/core/post_processing/bbox_nms.py", line 52, in multiclass_nms
    dets, keep = multiclass_nms_core(multi_bboxes, scores, score_thr, nms_cfg, max_num, return_inds)
  File "/usr/local/lib/python3.8/site-packages/mmdet/utils/deployment/symbolic.py", line 86, in wrapped_function
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/mmdet/core/post_processing/bbox_nms.py", line 67, in multiclass_nms_core
    bboxes = multi_bboxes.view(multi_scores.size(0), -1, 4)
```

@druzhkov-paul I've attached a snippet of the error log. As you can see, it fails at the end of the 1st epoch while running the evaluation, and an error is raised in reshaping (`bboxes = multi_bboxes.view(multi_scores.size(0), -1, 4)`) since `multi_bboxes` is empty

Reply:
https://github.com/openvinotoolkit/mmdetection/pull/342